spring-petclinic-customers-service/src/main/java/org/springframework/samples/petclinic/customers/web/PetDetails.java

```diff
@@ -15,36 +15,26 @@
  */
 package org.springframework.samples.petclinic.customers.web;
 
-import lombok.Data;
-
-import java.util.Date;
-
 import org.springframework.format.annotation.DateTimeFormat;
-import org.springframework.samples.petclinic.customers.model.Pet;
 import org.springframework.samples.petclinic.customers.model.PetType;
 
+import java.util.Date;
+
 /**
  * @author mszarlinski@bravurasolutions.com on 2016-12-05.
  */
-@Data
-class PetDetails {
 
-    private long id;
+record PetDetails(
 
-    private String name;
+    long id,
 
-    private String owner;
+    String name,
 
-    @DateTimeFormat(pattern = "yyyy-MM-dd")
-    private Date birthDate;
+    String owner,
 
-    private PetType type;
+    @DateTimeFormat(pattern = "yyyy-MM-dd")
+    Date birthDate,
 
-    PetDetails(Pet pet) {
```

Comment:
I think we could use custom record constructor here to avoid this kind of constructor call:
```
new PetDetails(pet.getId(), pet.getName(), pet.getOwner().getFirstName() + " " + pet.getOwner().getLastName(), pet.getBirthDate(), pet.getType());
```
See https://www.baeldung.com/java-records-custom-constructor

Reply:
The implicit constructor generated for record class by the compiler is used. 
I did not get the point whether  explicit constructor is required..

Reply:
I propose to add this constructor:
```
public PetDetails(Pet pet) {
       this(pet.getId(), pet.getName(), pet.getOwner().getFirstName() + " " + pet.getOwner().getLastName(), pet.getBirthDate(), pet.getType())
    }
```

Reply:
done